src/core/math/quat.js

```diff
@@ -104,14 +104,21 @@ class Quat {
      * Reports whether two quaternions are equal.
      *
      * @param {Quat} rhs - The quaternion to be compared against.
+     * @param {number} [epsilon] - The maximum difference between each component of the two quaternions.
      * @returns {boolean} True if the quaternions are equal and false otherwise.
      * @example
      * var a = new pc.Quat();
      * var b = new pc.Quat();
      * console.log("The two quaternions are " + (a.equals(b) ? "equal" : "different"));
      */
-    equals(rhs) {
-        return ((this.x === rhs.x) && (this.y === rhs.y) && (this.z === rhs.z) && (this.w === rhs.w));
+    equals(rhs, epsilon) {
```

Comment:
I'd vote for moving the version with epsilon to equalsAprox, and leave original function as is.

Reply:
What about `closeTo`?

Reply:
closeTo is not something you'd search for, so you'd likely just miss it.

Reply:
I wouldn't search for approx either.

Reply:
The only good thing is that `equalsApprox` would be next to `equals` in the API ref.

Reply:
yeah, and when you type `quad.equ...`, IntelliSense would offer both.

Reply:
Added the new `equalsApprox` method.